code-challenges.test.js

```diff
@@ -57,6 +133,60 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
-
+// declare a test function called fullHouse that takes in an array
+describe ("fullHouse", () => {
+    // write a test method that describes what the function is testing
+    // the test should look at each value in the array to determine if the array is a full house
+    it ("takes in an array of 5 numbers and determines whether or not the array has exactly one pair and one three of a kind", () => {
+        // write an expect method for each input
+        // if the input array is a full house: has one pair and one three of a kind, return true 
+        // if the input is not a full house, return false
+        expect(fullHouse(hand1)).toEqual(true)
+        expect(fullHouse(hand2)).toEqual(false)
+        expect(fullHouse(hand3)).toEqual(false)
+
+    })
+})
 
 // b) Create the function that makes the test pass.
+
+// create a function that takes in an array
+// for a full house, you only need two unique cards
+// find how many times the first number in the array occurs
+// try to see how many times the first value occurs in the array
+// increment the count of the first number in the array for how ever many times it occurs
+// fine the second unique number in the array
+// find the next humber in the array that is not equal to the first number (second unique number)
+// if the count of the first unique number is not two or three return false
+// find the cound of the second unique number
+// if the count of this number is not 2 or 3 then return false 
+// if the count of the first unique number is 2 or 3 AND the count of the second unique number is 2 or 3 then return true 
+
+const fullHouse = (array) => {
+    let count = [0,0]
+    let secondNum = 0
+    for (i=0; i<5; i++){
+        if (array[0] == array[i]){
+            count[0]++
+        }
+        else {
+            secondNum = array[i]
+        }
+    }
+    if ((count[0] !== 2) && (count[0] !== 3)){
+        return false
+    }
+    for (i=0; i<5; i++){
+        if (array[i] === secondNum){
+            count[1]++
+        }
+    }
+    if ((count[1] !==2) && (count[1] !==3)){
+        return false
+    }
+    return true
+}
```

Comment:
I like what you are doing here! The logic of keeping track of repeating elements can get a little messy and it's a very valuable skill. This code is perfect ground for refactor if you find the time and want to push your brain. Well done!